### PR TITLE
test&ci: add test for get-plugin-metadata command & add test CI

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: .github/.codecov.yml [hunk 1]
```diff
@@ -2,4 +2,5 @@ coverage:
   status:
     project:
       default:
-        target: 70%
\ No newline at end of file
+        # Set non-blocking status checks
```
Comment (shizhMSFT) [inline .github/.codecov.yml:5]: We need an issue to add it back.

File: .github/workflows/workflow.yml
Comment (shizhMSFT) [inline .github/workflows/workflow.yml:1]: Can we rename the file name to `test.yml`? `workflow.yml` is too general.

File: Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
     <RootNamespace>Notation.Plugin.AzureKeyVault</RootNamespace>
+    <RuntimeIdentifier>linux-x64</RuntimeIdentifier>
```
Comment (shizhMSFT) [inline Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj:7]: Will this impact releasing other runtime bundles?

**Review by shizhMSFT — COMMENTED** [review]
How can we fix those lint warnings?

File: .github/.codecov.yml (hunk 1 above)
Comment (JeyJeyGao) [reply .github/.codecov.yml:6]: Added issue #98

File: .github/workflows/workflow.yml
Comment (JeyJeyGao) [reply .github/workflows/workflow.yml:1]: Updated.

File: Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj (hunk 2 above)
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj]: This doesn't impact releasing other runtime bundles, however, I removed it.
Also removed:
```xml
<PublishSingleFile>true</PublishSingleFile>
    <PublishTrimmed>true</PublishTrimmed>
    <SelfContained>true</SelfContained>
```
So the config will be added as build flags in Release CI script.

**Comment by JeyJeyGao** [comment]
> How can we fix those lint warnings?

I removed the `PublishTrimmed` from `.csproj`, so there is no warning now. This config will be added as a flag in CI release script, so we may confront the warning when we do the release. The binary works well, but I will try to find a way to remove the warning. I think this PR should no be blocked by the issue.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**